src/main/java/com/ericsson/ei/controller/FailedNotificationControllerImpl.java

```diff
@@ -60,27 +64,38 @@
     public ResponseEntity<?> getFailedNotifications(
             @RequestParam(value = "subscriptionName", required = true) final String subscriptionName,
             final HttpServletRequest httpRequest) {
-        ObjectMapper mapper = new ObjectMapper();
-        QueryResponse queryResponse = new QueryResponse();
-        QueryResponseEntity queryResponseEntity = new QueryResponseEntity();
-        try {
-            List<String> response = processMissedNotification.processQueryMissedNotification(subscriptionName);
-            if (!response.isEmpty()) {
-                queryResponseEntity = mapper.readValue(response.get(0), QueryResponseEntity.class);
+        Set<String> subscriptionNameList = new HashSet<>(
+                Arrays.asList(subscriptionName.split(",")));
+        JSONArray foundArray = new JSONArray();
+        List<String> notFoundList = new ArrayList<>();
+
+        subscriptionNameList.forEach(name -> {
+            try {
+                List<String> response = processMissedNotification.processQueryFailedNotification(
+                        name);
+                if (!response.isEmpty()) {
+                    JSONObject object = new JSONObject(response.get(0));
+                    foundArray.put(object);
+                    LOGGER.debug("Successfully fetched failed notification for subscription [{}]",
+                            name);
+                }
+            } catch (Exception e) {
```

Comment:
Please make try/catch the first and last thing in the method and break out the logic from them. This method does a lot of things and is quite big.

Reply:
I'm not sure how to do this since directly after a notification is fetched it's put in one of two arrays/lists. These are later used to return or log. My problem was that if break parts of it out then I need to send the lists as parameters or something but I also need both of them back. I can't send two things back. They can't be global either since this is an API and many requests could be made in parallel. Do you have any suggestion how to handle this problem with the lists?

Reply:
I've now moved it into another function and the object I return is a JSONObject since serialization didn't play nice with json strings.